src/main/java/io/tackle/pathfinder/model/AbstractEntity.java

```diff
@@ -0,0 +1,64 @@
+package io.tackle.pathfinder.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+import io.quarkus.security.identity.SecurityIdentity;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.PrePersist;
+import javax.persistence.PreUpdate;
+import java.time.Instant;
+import java.util.Objects;
+
+@MappedSuperclass
+public abstract class AbstractEntity extends PanacheEntity {
+    @JsonIgnore
+    @CreationTimestamp
+    @Column(updatable=false)
+    public Instant createTime;
+    
+    public String createUser;
+    
+    @JsonIgnore
+    @UpdateTimestamp
+    public Instant updateTime;
+    
+    public String updateUser;
+    
+    @JsonIgnore
+    @Column(name = "deleted", columnDefinition = "BOOLEAN NOT NULL DEFAULT FALSE")
+    public Boolean deleted = false;
+
+    @PrePersist
+    protected void onCreate() {
+        final String username = getUsername();
+        createUser = username;
+        updateUser = username;
+    }
+
+    @PreUpdate
+    protected void onUpdate() {
+        updateUser = getUsername();
+    }
+
+    private String getUsername() {
+        // based on the Quarkus issue
+        // https://github.com/quarkusio/quarkus/issues/6948#issuecomment-619872942
+        SecurityIdentity context = CDI.current().select(SecurityIdentity.class).get();
+        if (Objects.nonNull(context)) {
+            String username = context.getPrincipal().getName();
+            if (Objects.nonNull(username)) {
+                return username;
+            }
+        }
+        // since all service are authenticated, it should never get here
+        // so maybe worth evaluating to throw an exception? food for thoughts
+        return "";
+    }
```

Comment:
- From the comment you mentioned in the commented lines I've also read https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766184194 and https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766367218 and It makes me wonder how do we know this block of code is working? I didn't see any of the tests are checking this value.
- Also I might be wrong but https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766367218 mentions `I think Quarkus is like "you don't use the bean? I'll not pack it then" on buildtime`  so it makes me wonder if this block of code won't work specially in native mode

Reply:
I've just tested the native artifact and it seems that `SecurityIdentity context = CDI.current().select(SecurityIdentity.class).get();` works as expected... However, in order to verify this I needed to manually run the native artifact and then go to see the DB manually, In my opinion, throwing an exception makes more sense since we won't notice if something is not working as expected if we just use `return "";`

Reply:
> * From the comment you mentioned in the commented lines I've also read [quarkusio/quarkus#6948 (comment)](https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766184194) and [quarkusio/quarkus#6948 (comment)](https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766367218) and It makes me wonder how do we know this block of code is working? I didn't see any of the tests are checking this value.
This class is a copy from tackle-controls:AbstractEntity , with only the cange on the expression for the delete.

> * Also I might be wrong but [quarkusio/quarkus#6948 (comment)](https://github.com/quarkusio/quarkus/issues/6948#issuecomment-766367218) mentions `I think Quarkus is like "you don't use the bean? I'll not pack it then" on buildtime`  so it makes me wonder if this block of code won't work specially in native mode
I have not paid attention to this class , tbh as it's the same as in TackleControls.
The only change is for the delete expression. We can later discuss about touching or not the tackle-controls one and remove this class from here

Reply:
In fact, in order to avoid code not tested in this project, I will revert to the tackle-controls-rest  file, and leave the delete expression on the sql files